### ISSUE TITLE
Wrong redirection due to / missing for non index.md files

### ISSUE THREAD

**Issue by Caetan95** [ultrabug/mkdocs-static-i18n#30]
Hello. @mic4ael and I found a problem using nginx for serving the static files. The navigation items are created without the leading slash, which creates a redirect that in some cases might be problematic. For example, we end up on `http://localhost/topic1/` instead of `http://localhost:8080/topic1/`.

![image](https://user-images.githubusercontent.com/71437950/120475590-d5df4280-c3a9-11eb-8618-1f67600c5bf2.png)

We think that a `/` is missing, so nginx redirects to `/topic1` instead of `/topic1/` and it breaks.

![image](https://user-images.githubusercontent.com/71437950/120477035-91ed3d00-c3ab-11eb-87b3-c06c19fc93e7.png)

As you can see, for these sections that have an `index.md` they have the `/` and there is no problem with the redirections.

To sum up, the thing is that we think that it's necessary to have `/site/` instead of `/site` for all the files, so we can expect the same behavior both for default and non-default languages and also for those files which are not `index.md`.

Thank you in advance :)

**Comment by mic4ael** [comment]
I just wanted to mention that this can be easily fixed with proper nginx configuration. However, looking at the URLs that plain `mkdocs` generates I would say that making sure that all URLs have leading slash would make this plugin a bit more consistent with the core.

**Comment by ultrabug** [comment]
Thanks for reporting, I'm in a road trip right now and will be able to check this properly next Tuesday

mic4ael is right, we should be consistent with the core which means that I'm missing some structure tests that should catch this first, then I'll fix it!

Please bear with me

**Comment by mic4ael** [comment]
No problem, take your time :D

**Comment by Stanzilla** [comment]
I found something similar, on this page, https://docs.coregames.com/fr/ hover over the "Forums" link in the header nav. It shows `https://docs.coregames.com/forums.coregames.com` because in the source it is `href="https:/learn.coregames.com"` notice the missing slash. In my yaml it's properly defined as `- Forums: https://forums.coregames.com`

**Comment by ultrabug** [comment]
@Stanzilla sorry I don't get it?

@Caetan95 @mic4ael I'm gonna start by writing a test that fails to demonstrate the problem first.

**Comment by Stanzilla** [comment]
@ultrabug it turns `https://forums.coregames.com` into `https:/forums.coregames.com`

**Comment by ultrabug** [comment]
@mic4ael @Caetan95 I found the culprit, please review #34 as it contains #33 and must be merged first

**Comment by ultrabug** [comment]
@Stanzilla please open another issue with more details on how to reproduce this, it does not look related to this issue to me?

or try #34 and see if that helps?

thanks mate

**Comment by Caetan95** [comment]
@ultrabug I have tested #34 and I confirm that it's working perfectly, brilliant!

**Comment by ultrabug** [comment]
version 0.15 is out mates, thanks!
